kie-remote/kie-remote-services/src/main/java/org/kie/remote/services/rest/async/JobResultManager.java

```diff
@@ -162,19 +131,60 @@ public JaxbDeploymentJobResult getJob(String jobId) {
 
         return job;
     }
-   
-    /**
-     * Get the most recent job requested for a given deployment
-     * @param deploymentId The id of the deployment
-     * @return The {@link JaxbDeploymentJobResult} with the job information
-     */
-    public JaxbDeploymentJobResult getMostRecentJob(String deploymentId) {
-        logger.debug( "Getting most recent job for '{}'", deploymentId);
-        String jobId = deploymentIdMostRecentJobIdMap.get(deploymentId);
-        if( jobId != null ) { 
-            return getJob(jobId);
+
+    protected Object getItemFromRequestInput(String itemName, RequestInfo requestInfo) {
```

Comment:
This is just nitpicking, feel free to ignore this: 

I'd much rather see 1 method that has a class/type-based if clause at the end: 

    if( result instanceof CommandContext ) { 
      return ((CommandContext) result).getData(itemName);
    } else if ( result instanceof ExecutionResults ) { 
      return ((ExecutionResults) result).getData(itemName);
    }

    return null;

There's no reason to check whether or not `.getData(itemName)` is null or not, seeing as we're returning null no matter what. :)